app/models/user.rb

```diff
@@ -0,0 +1,32 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  enum role: [ :admin, :buyer, :seller, :rider ]  # Use positional arguments
+
+  has_many :products
+  has_many :reviews
+  has_one :cart
+
+  validates :username, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP, message: 'must be a valid email address' }
+  validates :phone, presence: true, numericality: { only_integer: true }
```

Comment:
What if a phone number start with zero(s)?

Reply:
Fixing it. Thanks.